app/src/main/java/com/android/bookswap/ui/components/EntriesListBookComponent.kt

```diff
@@ -0,0 +1,239 @@
+                      .padding(horizontal = HORIZONTAL_PADDING.dp),
+              value = title.value,
+              label = { Text(text = "Title*") })
+        }
+
+        // Genres Dropdown
+        item {
+          ExposedDropdownMenuBox(
+              modifier = Modifier.fillMaxWidth().padding(horizontal = HORIZONTAL_PADDING.dp),
+              expanded = expanded,
+              onExpandedChange = { expanded = !expanded }) {
+                FieldComponent(
+                    value = genres.value.joinToString { it.Genre },
+                    onValueChange = {},
+                    readOnly = true,
+                    label = { Text(text = "Genres*") },
+                    modifier =
+                        Modifier.menuAnchor()
+                            .fillMaxWidth()
+                            .testTag(C.Tag.BookEntryComp.genre_field))
+                ExposedDropdownMenu(
+                    expanded = expanded,
+                    onDismissRequest = { expanded = false },
+                    modifier = Modifier.fillMaxWidth().fillMaxHeight(0.5f)) {
+                      BookGenres.values().forEach { genre ->
+                        val isSelected = genres.value.contains(genre)
+                        DropdownMenuItem(
+                            text = {
+                              Row(modifier = Modifier.fillMaxWidth()) {
+                                Text(
+                                    text = genre.Genre,
+                                    modifier = Modifier.align(Alignment.CenterVertically))
+                                Spacer(modifier = Modifier.weight(1f))
+                                if (isSelected) {
+                                  Icon(
+                                      imageVector = Icons.Default.Check,
+                                      contentDescription = "Selected")
+                                }
+                              }
+                            },
+                            onClick = {
+                              if (isSelected) {
+                                genres.value -= genre
+                              } else {
+                                genres.value += genre
+                              }
+                            },
+                            modifier =
+                                Modifier.testTag(
+                                    C.Tag.BookEntryComp.genre_menu + "_${genre.Genre}"))
+                      }
+                    }
+              }
+        }
+
+        // Author Input Field
+        item {
+          FieldComponent(
+              onValueChange = { author.value = it },
+              modifier =
+                  Modifier.testTag(C.Tag.BookEntryComp.author_field)
+                      .fillMaxWidth()
+                      .padding(horizontal = HORIZONTAL_PADDING.dp),
+              value = author.value,
+              label = { Text(text = "Author*") })
```

Comment:
Here and in some other places, these are UI text so should really be imported from strings.xml